src/platform/packages/private/kbn-grid-layout/grid/grid_row/delete_grid_row_modal.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+import React from 'react';
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { GridLayoutStateManager } from '../types';
+import { deleteRow, movePanelsToRow } from '../utils/row_management';
+
+export const DeleteGridRowModal = ({
+  rowIndex,
+  gridLayoutStateManager,
+  setDeleteModalVisible,
+}: {
+  rowIndex: number;
+  gridLayoutStateManager: GridLayoutStateManager;
+  setDeleteModalVisible: (visible: boolean) => void;
+}) => {
+  return (
+    <EuiModal
+      onClose={() => {
+        setDeleteModalVisible(false);
+      }}
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>Delete section</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        {i18n.translate('kbnGridLayout.deleteGridRowModal.body', {
+          defaultMessage:
+            'Are you sure you want to remove this section and its {panelCount} {panelCount, plural, one {panel} other {panels}}?',
```

Comment:
We do not show this modal when the section has zero panels, so we do not need to prevent the "... and its 0 panels" case